### ISSUE TITLE
Ignore non-IPFS content paths in DNSLink records

### ISSUE THREAD

**Issue by martinheidegger** [ipfs/go-dnslink#14]
DNSLink currently returns the [first non `/dns`-prefixed dnslink](https://github.com/ipfs/go-dnslink/blob/0626e50cfeaf0cd304eb0017de996e8cacb08540/dnslink.go#L190) which means that per domain it is possible to use dnslink for only one "protocol". In order for dnslink to become useful for different decentralized protocols (hyper/dat, earthlink, etc.) it would need an option to specify a particular protocol to be looked for. Would you be open to a PR adding an option to this repo, defaulting to IPFS?

(Related issue on js-dnslink: [here](https://github.com/JChanceHud/dnslink/issues/5))

**Comment by aschmahmann** [comment]
Yes, a PR would be welcome. However, the PR should go into go-namesys as that is the more used library for dnslink at the moment.

**Comment by martinheidegger** [comment]
@aschmahmann Probably somewhere here: https://github.com/ipfs/go-namesys/blob/2454122ef30efa1f4f86ffab33ea34b05eb848fe/dns.go#L171-L178, right?

**Comment by lidel** [comment]
@martinheidegger yes, we want to skip records where does not start with `/ipns/` or `/ipfs/` (right now it errors). Adding `strings.HasPrefix`  check should do the trick.

We should  also interpret `/dnslink` as `/ipns` and try to resolve it to the final form.
